packages/config/src/ast-utils/astConfigHelpers.ts

```diff
@@ -41,15 +42,18 @@ export interface ASTComponentDefinitionConfig {
  * }
  */
 export function addComponentDefinition (config: ASTComponentDefinitionConfig): t.ObjectProperty {
-  return extractProperty(`
-    const toMerge = {
-      component: {
-        devServer: {
-          framework: ${config.framework ? `'${config.framework}'` : 'undefined'},
-          bundler: '${config.bundler}',
-        },
+  const componentConfig = JSON.stringify({
+    component: {
+      devServer: {
+        framework: config.framework,
+        bundler: config.bundler,
       },
-    }
+      specPattern: config.specPattern,
```

Comment:
Doesn't this mean we will be scaffolding `specPattern: undefined` for the majority of frameworks (except Angular?)

Why do we need to do this? The default is  `**/*.cy.{js,jsx,ts,tsx}`. Shouldn't that cover Angular,  too?

Reply:
See: https://github.com/cypress-io/cypress/pull/22897#issuecomment-1192948652
`undefiend` values get dropped when scaffolding out the config, but it is confusing that an `undefined` is equivalent to the default spec pattern. Will see if I can come up with something cleaner

Reply:
> Since we pass the specPattern into the generated tsconfig

I might be misunderstanding something - what does this mean? Where do we pass `specPattern` into their `tsconfig.json`?

Unfortunate that `tsconfig.json` does not support curly braces.

Reply:
This is an Angular dev-server implementation detail. Angular needs to know about all of the typescript files included in the compilation ahead of time, so we need to push the specPattern glob into a tsconfig that we generate and pass into the webpack configuration. [Code here](https://github.com/cypress-io/cypress/blob/84ee67ec9b5a7f4faee532f1185d9080cdcb7cd5/npm/webpack-dev-server/src/helpers/angularHandler.ts#L127)

Since the `tsconfig.includes` doesn't support curly braces, we need to have a stricter specPattern than the default, otherwise the compilation will fail.

Reply:
Tweaked the types in [b3f6d4c](https://github.com/cypress-io/cypress/pull/22897/commits/b3f6d4c9a060a094f9f710d6f2b2aaaff3cdd057)

Reply:
I think you linked the wrong commit, but sounds good.